### PR TITLE
Feature: Make currency denominations configurable.

### PR DESCRIPTION
**Pull request by shabananavas**
Issue: https://www.drupal.org/node/2862645

### CODE REVIEW

File: modules/currency_denomination/includes/commerce_pos_currency_denomination.admin.inc [hunk 1]
```diff
@@ -0,0 +1,359 @@
...
+    );
+
+    $form['commerce_pos_denomination_name'] = array(
+      '#type' => 'machine_name',
+      '#title' => t('Denomination Name'),
+      '#default_value' => '',
+      '#maxlength' => 25,
+      '#description' => t('A unique denomination name eg. "penny".'),
+      '#machine_name' => array(
+        'exists' => 'commerce_pos_currency_denomination_name_exists',
```
Comment (wilderuncat) [inline modules/currency_denomination/includes/commerce_pos_currency_denomination.admin.inc:266]: Nice 👍

File: modules/currency_denomination/includes/commerce_pos_currency_denomination.admin.inc [hunk 2]
```diff
@@ -0,0 +1,359 @@
...
+
+    return confirm_form(
+      $form,
+      t('Are you sure you want to delete currency denomination type @type?', array(
+        '@type' => $form_state['denomination_type_info']->currency_code,
+      )),
+      'admin/commerce/config/pos/denomination',
+      t('All denominations under this type will be deleted. This action cannot be undone.'),
+      t('Delete'),
+      t('Cancel')
```
Comment (wilderuncat) [inline modules/currency_denomination/includes/commerce_pos_currency_denomination.admin.inc:146]: Really minor but I've noticed this a few places, there should be a trailing comma for multi-line arrays.

File: modules/currency_denomination/includes/commerce_pos_currency_denomination.admin.inc [hunk 3]
```diff
@@ -0,0 +1,359 @@
...
+          'rows' => $rows,
+          'sticky' => TRUE,
+          'attributes' => array(
+            'class' => array('commerce-pos-current-denominations')
+          )
+        )
+      );
+    }
+    else {
+      $output .= t('<p>Currently, no denominations have been added for this denomination type.</p>');
```
Comment (wilderuncat) [inline modules/currency_denomination/includes/commerce_pos_currency_denomination.admin.inc]: You could use the 'empty' parameter for theme_table for this if you want to simplify some of the logic here: https://api.drupal.org/api/drupal/includes%21theme.inc/function/theme_table/7.x

File: modules/keypad/commerce_pos_keypad.info [hunk 4]
```diff
@@ -4,3 +4,4 @@ core = 7.x
 package = Commerce POS
 
 dependencies[] = commerce_pos
+dependencies[] = commerce_pos:commerce_pos_currency_denomination
```
Comment (wilderuncat) [inline modules/keypad/commerce_pos_keypad.info:7]: This is really cool. But according to Drupal.org, we need to also require Drupal 7.40 when we use this new format: https://www.drupal.org/docs/7/creating-custom-modules/writing-module-info-files-drupal-7x#dependencies

dependencies[] = drupal:system (>= 7.40)

**Review by wilderuncat — CHANGES_REQUESTED** [review]
Looks great overall. Nice and clean. Just one minor thing that I think blocks it.

File: modules/currency_denomination/includes/commerce_pos_currency_denomination.admin.inc (hunk 2 above)
Comment (shabananavas) [reply modules/currency_denomination/includes/commerce_pos_currency_denomination.admin.inc:146]: These are arguments for the confirm_form() function, so no comma needed here.

File: modules/currency_denomination/includes/commerce_pos_currency_denomination.admin.inc (hunk 3 above)
Comment (shabananavas) [reply modules/currency_denomination/includes/commerce_pos_currency_denomination.admin.inc]: Yeah, I missed that one. Thanks!

File: modules/keypad/commerce_pos_keypad.info (hunk 4 above)
Comment (shabananavas) [reply modules/keypad/commerce_pos_keypad.info:7]: Thanks, will add that. We need to add that for the other modules in the POS system as well, as they're missing it.

**Review by wilderuncat — APPROVED** [review]

**Comment by smmccabe** [comment]
Committed via d.o patch, thnx!

**Pull request closed without merging**